Add view hierarchy docs for react native (#6217)
getsentry/sentry-docs: src/platforms/common/enriching-events/viewhierarchy.mdx

View hierarchies appear in the "Attachments" tab, where you can view all attachments, as well as associated events. Click the event ID to open the **Issue Details** page of that specific event.

![Screenshots List Example](viewhierarchy-list-example.png)

<PlatformSection supported={["react-native"]}>

## View Hierarchy mapping to React Native components

View Hierarchy captured by the SDK represents the native user interface generated based on React Native components structure. Generally each React Element maps to a native component, but there are exception like [View Flattening](https://reactnative.dev/architecture/view-flattening) described in the React Native documentation.

> Part of View Hierarchy are also native components wrapping the app created by React Native.

Reviewer: ```suggestion
Native components wrapping the app (created by React Native), are another part of View Hierarchy.
```

### iOS

> The code snipped below show React Native Component and next to it native UIKit components as they will be included in the view hierarchy.
> 

Reviewer: ```suggestion
The code snippet below shows React Native and native UIKit components the way they would be included in the view hierarchy.

```

```javascript
                                 // UIWindow
                                 // RCTRootContentView
                                 // ...
function MyComponent() {
  return (
    <View testID='id'>           // RTCView - id
      <Text>Hello World!</Text>  // RCTTextView
    </View>
  );
}
```

### Android

> The code snipped below show React Native Component and next to it native Android components as they will be included in the view hierarchy.
> 

Reviewer: ```suggestion
The code snippet below shows React Native and native Android components the way they would be included in the view hierarchy.

```
